File: ClientDashboard_API/Entities/Trainer.cs
namespace ClientDashboard_API.Entities
{
    public class Trainer : UserBase
    {
        public string? BusinessName { get; set; }

        public decimal? AverageSessionPrice { get; set; }

        public string? WorkoutRetrievalApiKey { get; set; }

        public bool EmailVerified { get; set; } = false;

        public bool AutoWorkoutRetrieval { get; set; } = false;

        public bool AutoPaymentSetting { get; set; } = false;

        public string? DefaultCurrency { get; set; }

        public List<string> ExcludedNames { get; set; } = [];

        public List<Client> Clients { get; set; } = [];
    }
}

File: ClientDashboard_GUI/src/app/models/trainer.ts
import { Client } from "./client";
import { UserBase } from "./user-base";

export interface Trainer extends UserBase {
  businessName?: string,
  averageSessionPrice?: number,
  workoutRetrievalApiKey?: string,
  emailVerified: boolean,
  autoWorkoutRetrieval: boolean,
  AutoPaymentSetting: boolean,
  defaultCurrency?: string,
  excludedNames: string[],
  clients: Client[];
}